packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.base.tsx

```diff
@@ -163,7 +169,7 @@ export class ColorPickerBase extends BaseComponent<IColorPickerProps, IColorPick
     }
 
     let newColor: IColor | undefined;
-    if (isHex) {
+    if (isHex && newValue && newValue.length === 6) {
```

Comment:
Would probably be better to do the isHex check like this, so we do nothing if the value is shorter than 6 chars (instead of re-updating back to the same color).
```
if (isHex) {
  if (newValue && newValue.length === 6) {
    ...
  }
} else {
  ...
}
```